### PR TITLE
Add CeedVectorViewRange utility

### PR DESCRIPTION
**Pull request by jeremylt**
@karenlstengel I think this is a version of what we were talking about on Friday?

### CODE REVIEW

File: interface/ceed-vector.c [hunk 1]
```diff
@@ -940,27 +940,46 @@ int CeedVectorReciprocal(CeedVector vec) {
 /**
   @brief View a CeedVector
 
-  @param[in] vec    CeedVector to view
-  @param[in] fp_fmt Printing format
-  @param[in] stream Filestream to write to
+  @param[in] vec         CeedVector to view
+  @param[in] num_entries Number of CeedVector entries to view
```
Comment (jedbrown) [inline interface/ceed-vector.c:944]: Maybe `CeedVectorViewRange()` taking a range (start, end, step)?

Comment (karenlstengel) [reply interface/ceed-vector.c:944]: I like this a lot. 
I think this will help a lot with mid-development debugging/sanity checking

Comment (jeremylt) [reply interface/ceed-vector.c:944]: Good idea, I like that better

File: interface/ceed-vector.c [hunk 2]
```diff
@@ -940,27 +940,49 @@ int CeedVectorReciprocal(CeedVector vec) {
...
+  @param[in] fp_fmt      Printing format
+  @param[in] stream      Filestream to write to
 
   @return An error code: 0 - success, otherwise - failure
 
   @ref User
 **/
-int CeedVectorView(CeedVector vec, const char *fp_fmt, FILE *stream) {
+int CeedVectorViewRange(CeedVector vec, CeedSize start, CeedSize stop, CeedSize step, const char *fp_fmt, FILE *stream) {
```
Comment (jedbrown) [inline interface/ceed-vector.c:952]: Looks like docs need an update.

File: interface/ceed-vector.c [hunk 3]
```diff
@@ -940,27 +940,49 @@ int CeedVectorReciprocal(CeedVector vec) {
...
   const CeedScalar *x;
   CeedCall(CeedVectorGetArrayRead(vec, CEED_MEM_HOST, &x));
 
   char fmt[1024];
   fprintf(stream, "CeedVector length %ld\n", (long)vec->length);
+  if (start != 0 || stop != vec->length || step != 1) {
+    fprintf(stream, "  start: %ld\n  stop:  %ld\n  step:  %ld\n", (long)start, (long)stop, (long)step);
+  }
+  if (stop > vec->length) stop = vec->length;
```
Comment (jedbrown) [inline interface/ceed-vector.c:961]: Will we allow negative steps? Should we document that one can safely use `INT_MAX`?

File: interface/ceed-vector.c (hunk 2 above)
Comment (jeremylt) [reply interface/ceed-vector.c]: oops, fixed

File: interface/ceed-vector.c (hunk 3 above)
Comment (jeremylt) [reply interface/ceed-vector.c:966]: I added a note that only the subset of the range that's valid for the vector will be printed.

I wasn't thinking of using negative steps but we can add it if you anticipate it being a used.

Comment (jeremylt) [reply interface/ceed-vector.c:968]: Good point wrt Python's behavior. Adjusted

**Review by jrwrigh — APPROVED** [review]

**Pull request merged**